draft-ietf-ppm-dap.md

```diff
@@ -2653,59 +2638,57 @@ running aggregation jobs ({{aggregate-flow}}) until the Collector initiates a
 collection job. This is because, in general, the aggregation parameter is not
 known until this point. In certain situations it is possible to predict the
 aggregation parameter in advance. For example, for Prio3 the only valid
-aggregation parameter is the empty string. For these reasons, the collection job
-is handled asynchronously. If aggregation is performed eagerly, the Leader MUST
-validate that the aggregation parameter received in the `CollectionJobReq`
-matches the aggregation parameter used in aggregations.
+aggregation parameter is the empty string.
 
 Upon receipt of a `CollectionJobReq`, the Leader begins by checking that it
-recognizes the task ID in the request path. If not, it MUST abort with error
-`unrecognizedTask`.
+recognizes the task ID. If not, it MUST abort with error `unrecognizedTask`.
+
+Next, the Leader checks that the indicated batch mode matches the task's batch
+mode. If not, then the Leader MUST abort with error `invalidMessage`.
 
 Next, the Leader checks that the aggregation parameter is valid as described in
-{{agg-param-validation}}. If the aggregation parameter is invalid, then the
-Leader MUST abort with error `invalidAggregationParameter`.
+{{agg-param-validation}}. If not, then the Leader MUST abort with error
+`invalidAggregationParameter`.
+
+If aggregation is performed eagerly, the Leader MUST validate that the
+aggregation parameter received in the `CollectionJobReq` matches the aggregation
+parameter used in aggregations.
+
+Changing a collection job's parameters is illegal, so further PUT requests to
+the collection job with a different `CollectionJobReq` MUST fail.
+
+After receiving the `CollectionJobReq`, the Leader begins working with the
+Helper to aggregate the reports satisfying the query (or continues this process,
+depending on the VDAF) as described in {{aggregate-flow}}.
 
-The Leader MAY further validate the request according to the requirements in
-{{batch-validation}} and abort with the indicated error, though some conditions
-such as the number of valid reports may not be verifiable while handling the
-`CollectionJobReq` message, and the batch will have to be re-validated later on
-regardless.
+The Leader first checks whether it can construct a valid batch for the
+collection job by applying the requirements in {{batch-validation}}. If so, then
+the Leader obtains the Helper's aggregate share following the aggregate-share
+request flow described in {{collect-aggregate}}. If not, it either aborts the
+collection job or tries again later, depending on which requirement in
```

Comment:
By "abort the collection job" do you mean abort the PUT request from the Collector?

Reply:
The former. The Leader's validation of  doesn't necessarily happen synchronously during handling of the PUT from the Collector. So I think we should avoid the verb "abort" here. I'll be more explicit here and reuse the language from later where we describe what to do if getting the aggregate share fails.